contracts/adapters/AaveLeverageStrategyExtension.sol

```diff
@@ -0,0 +1,1275 @@
+    }
+
+    /**
+     * Calculate total notional rebalance quantity and chunked rebalance quantity in collateral units. 
+     *
+     * return uint256          Chunked rebalance notional in collateral units
+     * return uint256          Total rebalance notional in collateral units
+     */
+    function _calculateChunkRebalanceNotional(
+        LeverageInfo memory _leverageInfo,
+        uint256 _newLeverageRatio,
+        bool _isLever
+    )
+        internal
+        view
+        returns (uint256, uint256)
+    {
+        // Calculate absolute value of difference between new and current leverage ratio
+        uint256 leverageRatioDifference = _isLever ? _newLeverageRatio.sub(_leverageInfo.currentLeverageRatio) : _leverageInfo.currentLeverageRatio.sub(_newLeverageRatio);
+
+        uint256 totalRebalanceNotional = leverageRatioDifference.preciseDiv(_leverageInfo.currentLeverageRatio).preciseMul(_leverageInfo.action.collateralBalance);
+
+        uint256 maxBorrow = _calculateMaxBorrowCollateral(_leverageInfo.action, _isLever);
+
+        uint256 chunkRebalanceNotional = Math.min(Math.min(maxBorrow, totalRebalanceNotional), _leverageInfo.twapMaxTradeSize);
+
+        return (chunkRebalanceNotional, totalRebalanceNotional);
+    }
+
+    /**
+     * Calculate the max borrow / repay amount allowed in precise units for lever / delever. This is due to overcollateralization requirements on
+     * assets deposited in lending protocols for borrowing.
+     * 
+     * For lever, max borrow is calculated as:
+     * (Net borrow limit in USD - existing borrow value in USD) / collateral asset price adjusted for decimals
+     *
+     * For delever, max repay is calculated as:
+     * Collateral balance in precise units * (net borrow limit in USD - existing borrow value in USD) / net borrow limit in USD
+     *
+     * Net borrow limit for levering is calculated as:
+     * The collateral value in USD * Aave collateral factor * (1 - unutilized leverage %)
+     *
+     * Net repay limit for delevering is calculated as:
+     * The collateral value in USD * Aave liquiditon threshold * (1 - unutilized leverage %)
+     *
+     * return uint256          Max borrow notional denominated in collateral asset
+     */
+    function _calculateMaxBorrowCollateral(ActionInfo memory _actionInfo, bool _isLever) internal view returns(uint256) {
+        
+        // Retrieve collateral factor and liquidation threshold for the collateral asset in precise units (1e16 = 1%)
+        ( , uint256 maxLtvRaw, uint256 liquidationThresholdRaw, , , , , , ,) = strategy.aaveProtocolDataProvider.getReserveConfigurationData(address(strategy.collateralAsset));
+
+        // Normalize LTV and liquidation threshold to precise units. LTV is measured in 4 decimals in Aave which is why we must multiply by 1e14
+        uint256 maxLtv = maxLtvRaw.mul(10 ** 14);
+        uint256 liquidationThreshold = liquidationThresholdRaw.mul(10 ** 14);
```

Comment:
We can remove this variable and usd `liquidationThresholdRaw.mul(10 ** 14)` directly in the `else` statement